traffic_ops/traffic_ops_golang/riak.go

```diff
@@ -0,0 +1,204 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/apache/incubator-trafficcontrol/lib/go-log"
+	"github.com/apache/incubator-trafficcontrol/traffic_ops/tostructs"
+	"github.com/basho/riak-go-client"
+	"github.com/jmoiron/sqlx"
+	"github.com/lestrrat/go-jwx/jwk"
+	"io/ioutil"
+	"net/http"
+)
+
+const riak_port = 8087
+const cdn_uri_keys_bucket = "cdn_uri_sig_keys" // riak namespace for cdn uri signing keys.
+
+func getStringValue(resp *riak.FetchValueResponse) (string, error) {
+	var obj *riak.Object
+
+	if len(resp.Values) > 0 {
+		obj = resp.Values[0]
+	} else {
+		return "", fmt.Errorf("no such object")
+	}
+	return string(obj.Value), nil
+}
+
+func assignDeliveryServiceUriKeysKeysHandler(db *sqlx.DB, cfg Config) AuthRegexHandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request, p PathParams, username string, privLevel int) {
```

Comment:
Why are we using `withData` if we don't need differentiation by `privLevel` on data handling?

Reply:
Will need the username and privLevel as I need to work on the tenancy checks.